placement/algo/sharded_helper.go

```diff
@@ -88,15 +88,15 @@ type PlacementHelper interface {
 	// Instances returns the list of instances managed by the PlacementHelper.
 	Instances() []placement.Instance
 
-	// HasRackConflict checks if the rack constraint is violated when moving the shard to the target rack.
-	HasRackConflict(shard uint32, from placement.Instance, toRack string) bool
+	// CanMoveShard checks if the shard can be moved from the instance to the target isolation group.
+	CanMoveShard(shard uint32, fromInstance placement.Instance, toIsolationGroup string) bool
 }
 
 type helper struct {
 	targetLoad         map[string]int
 	shardToInstanceMap map[uint32]map[placement.Instance]struct{}
-	rackToInstancesMap map[string]map[placement.Instance]struct{}
-	rackToWeightMap    map[string]uint32
+	igToInstancesMap   map[string]map[placement.Instance]struct{}
```

Comment:
nit: would it be more clear to call it `groupToInstancesMap`? `ig` is a bit vague IMO.